earthmover/earthmover.py

```diff
@@ -88,23 +94,92 @@ def load_config_file(self) -> dict:
         :param: params
         :return:
         """
-        _env_backup = os.environ.copy()
-
-        # Load & parse config YAML (using modified environment vars)
-        os.environ.update(self.params)
-
+        
+        # pass 1: grab config.macros (if any) so Jinja in the YAML can be
+        #   rendered with macros
+        with open(self.config_file, "r") as stream:
+            # cannot just yaml.load() here, since Jinja in the YAML may make it invalid...
+            # instead, pull out just the `config` section, which must not contain Jinja
+            # (except for `macros`) - then we yaml.load() just the config section to grab
+            # any `macros``
+            lines = stream.readlines()
+            start = -1
+            end = -1
+            macros = ""
+            for i in range(0,len(lines)):
+                if lines[i].strip()=="config:":
+                    start = i
+                    continue
+                if start>=0 and (lines[i].strip()=="" or (lines[i][0:2]!="  " and lines[i][0]!="\t")):

```

Comment:
This will break if the user puts a newline anywhere in the configs block. We actually want to parse until we get to the next top-level field. I'm building a workaround.